packages/strapi-admin/services/auth.js

```diff
@@ -34,9 +17,34 @@ const hashPassword = password => bcrypt.hash(password, 10);
  */
 const validatePassword = (password, hash) => bcrypt.compare(password, hash);
 
+/**
+ * Check login credentials
+ * @param {Object} options
+ * @param {string} options.email
+ * @param {string} options.password
+ */
+const checkCredentials = async ({ email, password }) => {
+  const user = await strapi.query('user', 'admin').findOne({ email });
+
+  if (!user) {
+    return [null, false, { message: 'Invalid credentials' }];
+  }
+
+  const isValid = await validatePassword(password, user.password);
+
+  if (!isValid) {
+    return [null, false, { message: 'Invalid credentials' }];
+  }
+
+  if (!(user.isActive === true)) {
+    return [null, false, { message: 'User not active' }];
+  }
+
+  return [null, user];
```

Comment:
Simple curiosity: what does this null stand for? (From what I saw it's used in the auth middleware, but since the value will always be null I wonder what's it used for)

Reply:
the three params stand for (error, user, info) it is used internally the passport local strategy. 
We pass an error only when we catch an error when call checkCredentials. We could simplify and remove this from the function